### PR TITLE
[Install] Document problem with Opam and Dune 2.x caching in OSX

### LINKED ISSUE #15138: Cannot install Coq 8.14 via opam on macOS due to sandboxing + dune 2.x caching
#### Description of the problem

Running `opam install coq.8.14.0` with a fresh opam install on macOS 11.6 fails for me:

```
[ERROR] The compilation of coq.8.14.0 failed at "./configure -configdir /Users/tchajed/.opam/default/lib/coq/config
        -prefix /Users/tchajed/.opam/default -mandir /Users/tchajed/.opam/default/man -docdir
        /Users/tchajed/.opam/default/doc/coq -libdir /Users/tchajed/.opam/default/lib/coq -datadir
        /Users/tchajed/.opam/default/share/coq -coqide no -native-compiler no".

#=== ERROR while compiling coq.8.14.0 =========================================#
# context     2.1.0 | macos/x86_64 | ocaml.4.12.0 | https://opam.ocaml.org#c7b1bb0d
# path        ~/.opam/default/.opam-switch/build/coq.8.14.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build ./configure -configdir /Users/tchajed/.opam/default/lib/coq/config -prefix /Users/tchajed/.opam/default -mandir /Users/tchajed/.opam/default/man -docdir /Users/tchajed/.opam/default/doc/coq -libdir /Users/tchajed/.opam/default/lib/coq -datadir /Users/tchajed/.opam/default/share/coq -coqide no -native-compiler no
# exit-code   1
# env-file    ~/.opam/log/coq-31916-bb6357.env
# output-file ~/.opam/log/coq-31916-bb6357.out
### output ###
# Error: too many retries (100): waiting for beacon file
# "/Users/tchajed/.cache/dune/db/runtime/dune-cache-daemon/port" to be locked
# Error: empty output starting cache
```

If I start the dune cache manually with `dune cache start`, I get a slightly different error message:

```
# Error: connect: : Operation not permitted
```

`opam install coq.8.13.2` still works, and I can compile Coq from source, and Homebrew can still build the Coq bottle, so the issue seems to be a regression in Coq's opam build process specifically (maybe due to sandboxing?).

#### Coq Version

- opam 2.1.0
- dune 2.9.1
- coq 8.14.0
- macOS 11.6

### PR DESCRIPTION
**Pull request by ejgallego**
Closes #15138

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -116,3 +119,11 @@ like `COQLIB="some path"`, that is a variable name followed by `=` and
 a string that follows OCaml's escaping conventions. This feature can be
 used by installers of binary package to make Coq aware of its installation
 path.
+
+Known problems
+--------------
+
+- On OSX, when using Opam with sandboxing and Dune < 3.0 with caching
+  enabled, Coq may fail to install, due to Opam's sandboxing
+  restricting the permissions needed for Dune 2.x caching daemon, see
```
Comment (jfehrle) [inline INSTALL.md:128]:
```suggestion
  restricting the permissions needed for the Dune 2.x caching daemon.  See
```

**Comment by ejgallego** [comment]
Anyone assigning?

**Review by jfehrle — APPROVED** [review]
Looks OK, though I think people are much more likely to look at the install instructions in https://coq.inria.fr/download.  In general we should try to cover each topic (e.g. installation) in a single place and have the others refer to it.  Otherwise it's much too easy to forget to update all the places, or, worse, they begin to contradict each other.  But cleaning that up can wait for another day.  Doesn't help that they are in different repositories :-(

@coqbot: merge now

**Comment by Zimmi48** [comment]
> Looks OK, though I think people are much more likely to look at the install instructions in https://coq.inria.fr/download.

Indeed, and since this is not really an issue when building Coq from source manually, but when building Coq through opam, the canonical documentation for this is https://coq.inria.fr/opam-using.html (which is referenced from https://coq.inria.fr/download). `INSTALL.md` should probably be kept for only the populations who need to understand how to build Coq from sources without relying on a package manager, i.e., developers and people making Coq packages (those that need instructions that are directly part of the version-controlled Coq sources rather than on the website).

**Pull request merged**